File: storydoc-backend/src/main/java/io/storydoc/server/ui/app/screendesign/SDComponentDTO.java
package io.storydoc.server.ui.app.screendesign;

import io.storydoc.server.ui.domain.screendesign.SDComponentId;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SDComponentDTO {
    private String name;
    private String type;
    private String label;
    private int x;
    private int y;
    private SDComponentId id;
    private List<ComponentAttributeDTO> attributes;
}

File: storydoc-frontend/src/app/api/models/sd-component-dto.ts
/* tslint:disable */
/* eslint-disable */
import { ComponentAttributeDto } from './component-attribute-dto';
import { SdComponentId } from './sd-component-id';
export interface SdComponentDto {
  attributes?: Array<ComponentAttributeDto>;
  id?: SdComponentId;
  label?: string;
  name?: string;
  type?: string;
  'x'?: number;
  'y'?: number;
}
